### PR TITLE
cpu/msp430: added periph UART driver

### PR DESCRIPTION
**Pull request by haukepetersen**
EDIT
~~depends on #3724~~

Finally also the UART driver for the MSP430 based boards.

While serial output (`hello-world`) is working fine, the shell is still having trouble. I am still debugging some unwanted crashing when returning from the UART RX interrupt. But since I am now to MSP430 devices, I PR this anyway, maybe someone experiences with this platform has an idea?

By the way: the default example does not work out of the box, as it is trying to use `uart0`, while I basically removed `uart0` support for these boards...

### CODE REVIEW

**Comment by kaspar030** [comment]
we're getting there! ;)

**Comment by OlegHahm** [comment]
Please remind me to test this on Monday.

**Comment by PeterKietzmann** [comment]
@OlegHahm, please test this today. It is monday :-)

**Comment by OlegHahm** [comment]
It's not Monday any more, but I just realized that I don't have any MSP430 flashing tools any more. Can't test until someone returns hers or his to me.

**Comment by kaspar030** [comment]
I'm on this now. TX seems fine, but I can reproduce crashes on RX.

**Comment by haukepetersen** [comment]
@kaspar030: Regarding the RX crashes, I just found similar behavior debugging the GPIO external interrupts. It seems to me, like there is something in the context switching that is broken for MSP430 platforms (but not sure yet...).

File: cpu/msp430fxyz/periph/uart.c [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    }
+
+    /* get the default UART for now -> TODO: enable for multiple devices */
+    msp_usart_t *dev = UART_DEV;
+
+    /* power off and reset device */
+    uart_poweroff(uart);
+    dev->CTL = USART_CTL_SWRST;
+    /* configure to 8N1 and using the SMCLK*/
+    dev->CTL = USART_CTL_CHAR;
```
Comment (thomaseichinger) [inline cpu/msp430fxyz/periph/uart.c]: This resets the reset bit which is holt while existing configuration.

Comment (haukepetersen) [inline cpu/msp430fxyz/periph/uart.c]: ups :-)

**Comment by haukepetersen** [comment]
rebased on changes in #3724 and fixed the interrupt problem. The `default` example (without uart0) does run smoothely now on the `msb-430(h)` boards.

**Comment by kaspar030** [comment]
@haukepetersen Nice, so this one is the only one standing between remove_uart0 and simplify_shell.

**Comment by kaspar030** [comment]
I can confirm that periph/uart works on msb-430h.

**Comment by haukepetersen** [comment]
fixed interrupt vector names for the wsn430 boards

**Comment by OlegHahm** [comment]
Confirmed working on wsn430-v1_3b, wsn430-v1_4, and Zolertia Z1.

**Comment by OlegHahm** [comment]
ACK

**Comment by haukepetersen** [comment]
rebased as #3724 was merged. Lets wait for Travis...

**Comment by haukepetersen** [comment]
oh happy day -> and go.

**Comment by OlegHahm** [comment]
`cd pub && more beer`

**Pull request merged**